### ISSUE TITLE
$1 param got ignored in createQueryBuilder update

### ISSUE THREAD

**Issue by BondWong** [typeorm/typeorm#2850]
**Issue type:**

[x] bug report

**Database system/driver:**

[x] `postgres`

**TypeORM version:**

[x] `latest`

**Steps to reproduce or a small repository showing the problem:**

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `postgres`
!>

**Comment by BondWong** [comment]
I am trying to create an updateQuery and execute, but something really strange happen.

The codes:
```
const updateQuery = getConnection(ctx.dbConnectionName)
            .createQueryBuilder()
            .update(child.tableName)
            .set({ _deletedAt: now })
            .where(`${child.foreignKeyColumnName} IN (:...ids)`, { ids });
          console.log(updateQuery.getSql());
          await updateQuery.execute();
```

The errors:
```
message: 'column "_deletedAt" is of type timestamp without time zone but expression is of type uuid',
    parameters: [
      Date 2018-09-26 04:18:43 982ms UTC {},
      '3d5f5dbd-6cf4-4360-a529-ad9208733592',
    ],
    position: '38',
    query: 'UPDATE "employer" SET "_deletedAt" = $2, "updatedAt" = CURRENT_TIMESTAMP WHERE applicant_id IN ($2)',
```

you can see that the _deletedAt is set to use $2 ?? why not $1???

**Comment by bmax** [comment]
Hello,

I believe I have more information on this bug.

The code below also shows the same problem but I think with a little more context/clarity:

```              
  const userRepository = connection.getRepository(User);
                const one_column = "new_update";
                const two_column = "hello";
                const qb = userRepository
                    .createQueryBuilder()
                    .update("layout_section_tables")
                    .set({
                        one_column
                    })
                    .where("two_column = :two_column", {
                        two_column
                    });
                console.log(qb.getSql());
                console.log(qb.getSql());
```

The console log returns the following:
```
UPDATE "layout_section_tables" SET "one_column" = $1 WHERE "two_column" = $2
UPDATE "layout_section_tables" SET "one_column" = $2 WHERE "two_column" = $2
```

I have identified the problem but not sure if my fix is correct.

Here is where the problem lies: https://github.com/typeorm/typeorm/blob/master/src/query-builder/UpdateQueryBuilder.ts#L390

My solution is adding `this.connection.driver instanceof PostgresDriver ||`
to the ternary.

I can create the PR if you think that is the best way to go.
cc @Kononnable @pleerock

**Comment by imnotjames** [comment]
This should be fixed with #7821
